.pipelines/templates/.builder-release-template.yaml

```diff
@@ -1,13 +1,38 @@
 parameters:
-    - name: artifactName
-      type: string
-      default: 1604
-
+  - name: artifactName
+    type: string
+    default: 1604
+  - name: useOverrides
+    type: boolean
+    default: false
+  - name: overrideBranch
+    type: string
+    default: master
 steps:
+  - checkout: self
+    # s is the default path for repositories - if we don't set this when using multiple repsositories, then it is the repo name
+    # "s/AgentBaker", but most of the pipeline variables still refer to "s" and many of the scripts below fail.
+    path: s
+    fetchTags: false
+    fetchDepth: 1
+  - checkout: git://CloudNativeCompute/aks-rp@${{ parameters.overrideBranch }}
+    path: aks-rp
+    fetchDepth: 1
+    fetchTags: false
+    lfs: false
+    condition: eq('${{ parameters.useOverrides }}', true)
   - bash: |
-        m="linuxVhdMode" && \
-        echo "Set build mode to $m" && \
-        echo "##vso[task.setvariable variable=MODE]$m"
+      set -ex
+      # use rsync rather than cp because it worked when I tested this script. And it's output is pretty.
+      # debated using -u flag here to only copy newer files, but decided that if there are overrides, then
+      # we should use them!
+      rsync -av $(Pipeline.Workspace)/aks-rp/agentbakersvc/overrides/ ./
```

Comment:
let's make a note somewhere that the `agentbakersvc/overrides` is meant to contain files/directories relative to the AgentBaker repo root

Reply:
this is also where we'd need to be careful about replace vs. merge, right now we're just blindly replacing

Reply:
I think blind replace would pretty much make sense for any file other than components.json

though we also need to think about potential edge cases here - the embargoed VHDs built with overrides will still be bootstrapped with the latest (1P/3P available) CSE/customdata returned by latest agentbakersvc, what if we somehow introduce an issue where the overrides contain a version of a file that is also used at provisioning time (such as cse_install.sh, renamined to provision_install.sh during the build) which is incompatible? (e.g. renaming a function that cse_main.sh relies on, etc.)

Reply:
with that in mind ^, I'm wondering if only allowing components.json replace/merge would be prudent

Reply:
I think you're right. Have changed to only replace components.json. Decided on replace vs merge because predicting the results of a merge can be hard, and if we need to do the override, whoever does it can just copy the entire existing components.json and edit it.